src/app/components/tableTranslation/TableTranslation.tsx

```diff
@@ -0,0 +1,92 @@
+"use client";
+import { createTranslatProject } from '@/app/service/DictionariesServices';
+import { getAllProjects } from '@/app/service/ProjectServices';
+import { schemaCreateTranslation } from '@/app/utils/TranslationValidation';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Box, Button, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, TextField, Typography } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { toast } from 'react-toastify';
+import styles from "./style.module.scss";
+
+interface TranslatData {
+  key: string;
+  text: string;
+}
+
+const TableTranslation = ({ projectsId }: { projectsId: any }) => {
+  const { data: Allprojects} = useQuery('Allprojects', getAllProjects);
+
+  const queryClient = useQueryClient();
+  const { register, handleSubmit , reset} = useForm<TranslatData>({
+    resolver: yupResolver(schemaCreateTranslation),
+    defaultValues: {
+      key: '',
+      text: '',
+    },
+  });
+
+  const handleTranslation = (data: TranslatData,) => createTranslatProject(data,projectsId);
+
+   const { mutate: createTranslation } = useMutation(handleTranslation, {
```

Comment:
don't pass argument like that , destructure all, and in use query also destructure it , like that: 
```
const {data, isLoading, error} = useQuery({
 queryKey: ['allProjects'],
 queryFn: getAllProjects,
})
```

this is more readable and maintainable
you should think that may be someone read your code don't know any think about react query, so make the code more readable so if he need search , he will be know what to search about.